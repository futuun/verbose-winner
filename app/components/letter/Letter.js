import React, { PropTypes } from 'react'
import classNames from 'classnames'
import styles from './styles.sass'

function Letter(props) {
  const {
    value,
    exposed,
    inactive,
  } = props

  const letterClass = classNames(
    styles.letter,
    { [styles.inactive]: inactive },
  )

  return (
    <div className={letterClass}>
      {exposed ? value : null}
    </div>
  )
}

Letter.defaultProps = {
  value: '',
  exposed: false,
  inactive: false,
}

Letter.propTypes = {
  value: PropTypes.string.isRequired,
  exposed: PropTypes.bool.isRequired,
  inactive: PropTypes.bool.isRequired,
}

export default Letter
