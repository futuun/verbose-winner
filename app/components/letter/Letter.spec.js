import React from 'react'
import renderer from 'react-test-renderer'
import Letter from './Letter'

test('Letter: unexposed, active', () => {
  const component = renderer.create(
    <Letter
      value="A"
      exposed={false}
      inactive={false}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Letter: exposed, active', () => {
  const component = renderer.create(
    <Letter
      value="A"
      exposed
      inactive={false}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Letter: inactive', () => {
  const component = renderer.create(
    <Letter inactive />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
