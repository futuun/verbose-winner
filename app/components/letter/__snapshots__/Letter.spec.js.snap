exports[`test Letter: exposed, active 1`] = `
<div
  className="letter">
  A
</div>
`;

exports[`test Letter: inactive 1`] = `
<div
  className="letter inactive" />
`;

exports[`test Letter: unexposed, active 1`] = `
<div
  className="letter" />
`;
